src/sql/statement.rs

```diff
@@ -1190,7 +1222,7 @@ fn handle_create_source(scx: &StatementContext, stmt: Statement) -> Result<Plan,
                         path: path.clone().into(),
                         tail,
                     });
-                    let encoding = get_encoding(with_options)?;
+                    let encoding = get_encoding()?;
                     (connector, encoding)
                 }
                 Connector::AvroOcf { path, .. } => {
```

Comment:
To follow up on my comment above: I was expecting this to work like it does for views, where we simply fix up the `RelationDesc` with the new column names after we've generated the source. This is the hunk of code I mean: https://github.com/MaterializeInc/materialize/blob/fd8f87b5025ce01d43c8efd5d0a39f4821098641/src/sql/statement.rs#L845-L857

I think that you could copy and paste that nearly directly on line 1228, below, and it would just work? (Or am I missing something)? That way it would also apply to all sources equally!